packages/snaps-controllers/src/snaps/endowments/caveats/requestTime.ts

```diff
@@ -0,0 +1,129 @@
+import { assertStruct, hasProperty, isPlainObject } from '@metamask/utils';
+
+import type { CaveatMapperFunction, CaveatMapperReturnValue } from './generic';
+
+/**
+ * Asserts that the given value is a valid `maxRequestTime` value.
+ *
+ * @param value - The value to assert.
+ * @param ErrorWrapper - An optional error wrapper to use. Defaults to
+ * {@link AssertionError}.
+ * @throws If the value is not a valid `maxRequestTime` value.
+ */
+function assertIsMaxRequestTime(
+  value: unknown, // eslint-disable-next-line @typescript-eslint/naming-convention
+  ErrorWrapper?: AssertionErrorConstructor,
+): asserts value is number {
+  assertStruct(
+    value,
+    MaxRequestTimeStruct,
+    'Invalid JSON-RPC origins',
+    ErrorWrapper,
+  );
+}
+
+/**
+ * Validate the value of a caveat. This does not validate the type of the
+ * caveat itself, only the value of the caveat.
+ *
+ * @param caveat - The caveat to validate.
+ * @throws If the caveat value is invalid.
+ */
+function validateMaxRequestTimeCaveat(caveat: Caveat<string, any>) {
+  if (!hasProperty(caveat, 'value') || !isPlainObject(caveat.value)) {
+    throw rpcErrors.invalidParams({
+      message: 'Invalid maxRequestTime caveat',
+    });
+  }
+
+  const { value } = caveat;
+  assertIsMaxRequestTime(value, rpcErrors.invalidParams);
+}
+
+/**
+ * Map a raw value from the `initialPermissions` to a caveat specification.
+ * Note that this function does not do any validation, that's handled by the
+ * PermissionsController when the permission is requested.
+ *
+ * @param value - The raw value from the `initialPermissions`.
+ * @returns The caveat specification.
+ */
+export function getMaxRequestTimeCaveatMapper(
+  value: Json,
+): CaveatMapperReturnValue {
+  return {
+    caveats: [
+      {
+        type: SnapCaveatType.MaxRequestTime,
+        value,
+      },
+    ],
+  };
+}
+
+/**
+ * Creates a wrapping caveat mapper that creates the `maxRequestTime` caveat
+ * and merges it with any other caveats created by the mapper function.
+ *
+ * @param mapper - Another caveat mapper function.
+ * @returns The caveat specification.
+ */
+export function createMaxRequestTimeMapper(
+  mapper: CaveatMapperFunction,
+): CaveatMapperFunction {
+  return function (value: Json) {
+    // We assume this to be used only with caveats of this type
+    const { maxRequestTime, ...rest } = value as Record<string, Json>;
+
+    const mapperResult = mapper(rest);
+
+    if (!maxRequestTime) {
+      return mapperResult;
+    }
+
+    return {
+      ...mapperResult,
+      caveats: [
+        ...(mapperResult.caveats ?? []),
+        {
+          type: SnapCaveatType.MaxRequestTime,
+          value: maxRequestTime,
+        },
+      ],
+    };
+  };
+}
+
+/**
+ * Getter function to get the {@link MaxRequestTime} caveat value from a permission if specified.
+ *
+ * @param permission - The permission to get the caveat value from.
+ * @returns The caveat value if present.
```

Comment:
Should probably say something about a return value if a caveat is not present.

Reply:
Fixed!